### ISSUE TITLE
Force objects/arrays/ArgumentLists/FormalParameters that can fit on one line, onto one line

### ISSUE THREAD

**Issue by graingert** [eslint/eslint#7526]
```js
//good
// eslint max-len=16
function foo(
            ham,
            spam,
            eggs,
) { ... }

// bad
function foo(ham, spam, eggs) { ... }

```

```js
// eslint max-len=100
//bad
function foo(
            ham,
            spam,
            eggs,
) { ... }

// good
function foo(ham, spam, eggs) { ... }
```

**Comment by graingert** [comment]
This is a rule request, excuse the lack of template I'm afk

**Comment by not-an-aardvark** [comment]
Thanks for the rule suggestion.

To clarify, how would this rule determine whether parameters can "fit" on one line? It looks like you're referencing the `max-len` rule, but ESLint rules aren't allowed to see the configuration of other rules (since it has to be possible to enable/disable rules independently).

**Comment by graingert** [comment]
Actually it's probably best to give it the number separately:

```js

const maxLen = 100;

...
  rules: {
    'max-len': [2, maxLen],
    'fit-on-one-line': [2, maxLen],
  }
...
```

**Comment by not-an-aardvark** [comment]
What about other things, such as comma and brace style? For example, the length of a line will be different depending on whether it's formatted like this:

```js
function foo(bar,bar,qux)
{
}
```

or like this:

```js
function foo ( bar, baz, qux ) {
}
```

I'm worried that having the rule depend on the *length* of a line has too many potential factors that would depend on other rules.

**Comment by graingert** [comment]
For mvp it could strip all unnecessary whitespace, that way you'd only get
false negatives

**Comment by graingert** [comment]
Eg.

```js
function foo (ham,                                          spam) { };
```

Is long enough to be split, but another rule will have to be used to catch it

**Comment by not-an-aardvark** [comment]
It looks like the proposal has changed a bit since the original issue, and I'm a bit confused about what options it would have.

May you please edit the original issue and include answers to the questions in the [rule proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md)?

**Comment by graingert** [comment]
Sure

**Comment by ljharb** [comment]
I thought about another case that should be allowed via the airbnb guide:
```js
foo(a, b, [
  c,
  d,
]);
```

**Comment by nzakas** [comment]
Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.

Remember, you can still create this rule for yourself and distribute it as a plugin.
